### PR TITLE
dts: arm: st: Remove use of CONFIG_SOC_* from STM32 F1 dts files

### PR DESCRIPTION
**Pull request by galak**
To move forward and remove use of Kconfig in dts files lets just create
SoC specific dtsi files that the boards can include.  We also seperate
out the F1 dtsi files into their own dir.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
@erwango What is the difference between stm32f103Xb & stm32f103X8?

The board stm32_min_dev has the Kconfig.board depend on SOC_STM32F103X8, but the .dts does #include <st/stm32f103Xb.dtsi>

**Comment by erwango** [comment]
> The board stm32_min_dev has the Kconfig.board depend on SOC_STM32F103X8, but the .dts does #include <st/stm32f103Xb.dtsi>

So this boards actually embeds a STM32F103C8T6.
Though, Cube SDK doesn't provide a CMSIS file for STM32F103x8 variants and we're supposed to use STM32F103xR as SoC reference. I think this is the reason why we ended up using st/stm32f103Xb.dtsi.
Though, both SoCs actually differ in term of abilities (number of A/D channels and high current I/Os) and we should rather have used a dedicated stm32f103X8.dtsi to be correct.

File: dts/arm/st/f1/stm32f103X8.dtsi [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017 I-SENSE group of ICCS
```
Comment (erwango) [inline dts/arm/st/f1/stm32f103X8.dtsi]: Use you own commit

File: dts/arm/st/f1/stm32f103X8.dtsi [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * SoC device tree include for STM32F103xB SoCs
```
Comment (erwango) [inline dts/arm/st/f1/stm32f103X8.dtsi]: STM32F103xB > STM32F103x8

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/f1/stm32f103X8.dtsi (hunk 2 above)
Comment (galak) [reply dts/arm/st/f1/stm32f103X8.dtsi]: dropped the comment block

**Comment by galak** [comment]
@erwango take another look.

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Pull request merged**